lib/data/multipleBackendGateway.js

```diff
@@ -48,23 +115,52 @@ const multipleBackendGateway = {
     },
 
     get: (objectGetInfo, range, reqUids, callback) => {
+        const key = objectGetInfo.key ? objectGetInfo.key : objectGetInfo;
         const client = clients[objectGetInfo.dataStoreName];
         if (client.clientType === 'scality') {
-            return client.get(objectGetInfo.key, range, reqUids, callback);
+            return client.get(key, range, reqUids, callback);
+        }
+        if (client.clientType === 'aws_s3') {
+            const stream = client.getObject({
+                Bucket: client.awsBucketName,
+                Key: key,
+                Range: range,
+            }).createReadStream().on('error', err => {
+                const log = createLogger(reqUids);
+                log.error('error creating stream', { err });
+                return callback(err);
```

Comment:
please make sure there are tests where you expect an error from aws -- like no such key, access denied.